packages/shared/src/loadClerkJsScript.ts

```diff
@@ -32,56 +33,142 @@ type LoadClerkJsScriptOptions = Without<ClerkOptions, 'isSatellite'> & {
+      clearTimeout(timeoutId);
+      clearInterval(pollInterval);
+    };
+
+    const checkAndResolve = () => {
+      if (resolved) return;
+
+      if (isClerkProperlyLoaded()) {
+        resolved = true;
+        cleanup(timeoutId, pollInterval);
+        resolve(null);
+      }
+    };
+
+    const handleTimeout = () => {
+      if (resolved) return;
+
+      resolved = true;
+      cleanup(timeoutId, pollInterval);
+
+      if (!isClerkProperlyLoaded()) {
+        createAndAssignFallbackClerk();
+        reject(new Error(FAILED_TO_LOAD_ERROR));
+      } else {
+        resolve(null);
+      }
+    };
+
+    const timeoutId = setTimeout(handleTimeout, timeoutMs);
+
+    checkAndResolve();
+
+    const pollInterval = setInterval(() => {
+      if (resolved) {
+        clearInterval(pollInterval);
+        return;
+      }
+      checkAndResolve();
+    }, 100);
+  });
+}
+
+/**
+ * Hotloads the Clerk JS script with robust failure detection.
  *
- * Checks for an existing Clerk JS script. If found, it returns a promise
- * that resolves when the script loads. If not found, it uses the provided options to
- * build the Clerk JS script URL and load the script.
+ * Uses a timeout-based approach to ensure absolute certainty about load success/failure.
+ * If the script fails to load within the timeout period, or loads but doesn't create
+ * a proper Clerk instance, a fallback instance with error status is created.
  *
  * @param opts - The options used to build the Clerk JS script URL and load the script.
  *               Must include a `publishableKey` if no existing script is found.
  *
  * @example
  * loadClerkJsScript({ publishableKey: 'pk_' });
  */
-const loadClerkJsScript = async (opts?: LoadClerkJsScriptOptions) => {
+const loadClerkJsScript = async (opts?: LoadClerkJsScriptOptions): Promise<HTMLScriptElement | null> => {
+  const timeout = opts?.scriptLoadTimeout ?? 15000;
+
+  if (isClerkProperlyLoaded()) {
+    return null;
+  }
+
   const existingScript = document.querySelector<HTMLScriptElement>('script[data-clerk-js-script]');
 
   if (existingScript) {
-    return new Promise((resolve, reject) => {
-      existingScript.addEventListener('load', () => {
-        resolve(existingScript);
-      });
-
-      existingScript.addEventListener('error', () => {
-        reject(FAILED_TO_LOAD_ERROR);
-      });
-    });
+    return waitForClerkWithTimeout(timeout);
   }
 
   if (!opts?.publishableKey) {
     errorThrower.throwMissingPublishableKeyError();
-    return;
+    return null;
   }
 
-  return loadScript(clerkJsScriptUrl(opts), {
+  const loadPromise = waitForClerkWithTimeout(timeout);
+
+  loadScript(clerkJsScriptUrl(opts), {
     async: true,
     crossOrigin: 'anonymous',
     nonce: opts.nonce,
     beforeLoad: applyClerkJsScriptAttributes(opts),
-  }).catch(() => {
-    throw new Error(FAILED_TO_LOAD_ERROR);
-  });
+  }).catch(() => {});
```

Comment:
if `loadScript` errors, wouldn't we want to know about it before the timeout expires?

Reply:
@jacekradko Agreed, we should handle an error from `loadScript()` immediately.